roles/ceph-facts/tasks/facts.yml

```
@@ -289,3 +289,8 @@
     set_fact:
       ntp_service_name: ntpd
     when: ansible_os_family in ['RedHat', 'Suse']
+
+- name: set grafana_server_addr fact
+  set_fact:
+    grafana_server_addr: "{{ hostvars[groups[grafana_server_group_name][0] | default(groups[mgr_group_name][0])]['ansible_all_ipv4_addresses'] | ipaddr(public_network) | first }}"
```

I think you need to add `()` here
```suggestion
    grafana_server_addr: "{{ (hostvars[groups[grafana_server_group_name][0] | default(groups[mgr_group_name][0])])['ansible_all_ipv4_addresses'] | ipaddr(public_network) | first }}"
```